src/components/discovery/DiscoverySearchForm.js

```diff
@@ -34,180 +33,180 @@ const VARIANT_OPTIONAL_FIELDS = [
     "[dataset item].reference",
 ];
 
-// noinspection JSUnusedGlobalSymbols
-const CONDITION_RULES = [
-    {
-        validator: (rule, value, cb) => {
-            if (value.field === undefined) {
-                cb("A field must be specified for this search condition.");
-            }
-
-            const searchValue = getSchemaTypeTransformer(value.fieldSchema.type)[1](value.searchValue);
-            const isEnum = value.fieldSchema.hasOwnProperty("enum");
-
-            // noinspection JSCheckFunctionSignatures
-            if (
-                !VARIANT_OPTIONAL_FIELDS.includes(value.field) &&
-                (searchValue === null ||
-                    (!isEnum && !searchValue) ||
-                    (isEnum && !value.fieldSchema.enum.includes(searchValue)))
-            ) {
-                cb(`This field is required: ${searchValue}`);
-            }
-
-            cb();
-        },
-    },
+const updateVariantConditions = (conditions, fieldName, searchValue) =>
+    conditions.map((c) =>
+        c.value.field === fieldName
+            ? {...c, value: { ...c.value, searchValue }}
+            : c);
 
-];
+const conditionValidator = (rule, { field, fieldSchema, searchValue }) => {
+    if (field === undefined) {
+        return Promise.reject("A field must be specified for this search condition.");
+    }
 
+    const transformedSearchValue = getSchemaTypeTransformer(fieldSchema.type)[1](searchValue);
+    const isEnum = fieldSchema.hasOwnProperty("enum");
+
+    // noinspection JSCheckFunctionSignatures
+    if (
+        !VARIANT_OPTIONAL_FIELDS.includes(field) &&
+        (transformedSearchValue === null ||
+            (!isEnum && !transformedSearchValue) ||
+            (isEnum && !fieldSchema.enum.includes(transformedSearchValue)))
+    ) {
+        return Promise.reject(`This field is must have a value: ${field}`);
```

Comment:
Phrasing

Reply:
Validation: we may want to prevent users from submitting white-space search values